pandas/io/stata.py

```diff
@@ -625,10 +625,11 @@ def _cast_to_stata_types(data: DataFrame) -> DataFrame:
                 if data[col].max() >= 2 ** 53 or data[col].min() <= -(2 ** 53):
                     ws = precision_loss_doc.format("int64", "float64")
         elif dtype in (np.float32, np.float64):
-            value = data[col].max()
+            value = np.abs(data[col]).max()
```

Comment:
Could you just check `if np.isinf(data[col]).any()`

Reply:
OK and it seems faster than calculate absolute value as well!